verilog/analysis/checkers/always_comb_blocking_rule.h

```diff
@@ -0,0 +1,69 @@
+// Copyright 2017-2019 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef VERIBLE_VERILOG_ANALYSIS_CHECKERS_ALWAYS_COMB_BLOCKING_RULE_H_
+#define VERIBLE_VERILOG_ANALYSIS_CHECKERS_ALWAYS_COMB_BLOCKING_RULE_H_
+
+#include <string>
+#include <vector>
+
+#include "common/analysis/lint_rule_status.h"
+#include "common/analysis/matcher/matcher.h"
+#include "common/analysis/matcher/matcher_builders.h"
+#include "common/analysis/syntax_tree_lint_rule.h"
+#include "common/text/symbol.h"
+#include "common/text/syntax_tree_context.h"
+#include "verilog/CST/verilog_matchers.h"  // IWYU pragma: keep
+#include "verilog/analysis/descriptions.h"
+
+namespace verilog {
+namespace analysis {
+
+class AlwaysCombBlockingRule : public verible::SyntaxTreeLintRule {
+ public:
+  using rule_type = verible::SyntaxTreeLintRule;
+  static absl::string_view Name();
+
+  // Returns the description of the rule implemented formatted for either the
+  // helper flag or markdown depending on the parameter type.
+  static std::string GetDescription(DescriptionType);
+
+  void HandleSymbol(const verible::Symbol& symbol,
+                    const verible::SyntaxTreeContext& context) override;
+
+  verible::LintRuleStatus Report() const override;
+
+ private:
+  // Link to style guide rule.
+  static const char kTopic[];
+
+  // Diagnostic message.
+  static const char kMessage[];
+
+  using Matcher = verible::matcher::Matcher;
+
+  const Matcher always_comb_matcher_ = NodekAlwaysStatement(
+      AlwaysCombKeyword(), HasNonblockingAssignmentStatement());
+  const Matcher always_comb_block_matcher_ = NodekAlwaysStatement(
+      AlwaysCombKeyword(), HasNonblockingAssignmentStatementInBlock());
+
+  std::vector<verible::LintViolation> violations_;
+
+  unsigned inside = 0;
```

Comment:
Is this used anywhere?  (if not, delete)

Reply:
Same as above. Removed.